static/js/SourceTranslationsButtons.jsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import PropTypes from "prop-types";
+import {InterfaceText} from "./Misc";
+
+function SourceTranslationsButtons(props) {
+    const showSource = props.showSource;
+    const showTranslation = props.showTranslation;
+    const setShowTexts = props.setShowTexts;
```

Comment:
if the parent component changes, without telling react to memoize the props it will rerender this component. To achieve that, use [useCallback](https://react.dev/reference/react/useCallback)